scripts/cluster-upgrade.sh

```diff
@@ -349,6 +347,68 @@ upgrade() {
         log $OCM_NAME "upgrade" "deleted openshift-machine-api/deployment/cluster-autoscaler-operator"
     fi
 
+    log $OCM_NAME "upgrade" "checking availability of ReplicaSets and DaemonSets"
+
+    # verify all replicasets are at expected counts
+    TOTAL_DESIRED=0
+    TOTAL_SCHEDULED=0
+    while [ $TOTAL_DESIRED -eq 0 ] || [ $TOTAL_DESIRED -ne $TOTAL_SCHEDULED ];
+    do
+        # don't sleep on first pass
+        if [ $TOTAL_DESIRED -gt 0 ];
+        then
+            sleep 15
+        fi
```

Comment:
Maybe on lines 353 the initial values can be initialized (cf lines 367-370, 372-375) to avoid the special case here?

Reply:
I could but it duplicates the whole loop.  I've pushed an update that doesn't duplicate things, moves the check into the loop, if/else.  Testing in stage to verify it's good.

Reply:
The update works, keeps it from progressing until all RS and DS are ready.